app/services/declarations/clawback.rb

```diff
@@ -0,0 +1,47 @@
+module Declarations
+  class Clawback
+    attr_reader :author, :declaration, :voided_by_user_id, :next_available_output_fee_statement
+
+    def initialize(author:, declaration:, next_available_output_fee_statement:, voided_by_user_id: nil)
+      @author = author
+      @declaration = declaration
+      @voided_by_user_id = voided_by_user_id
+      @next_available_output_fee_statement = next_available_output_fee_statement
+    end
+
+    def clawback
+      ActiveRecord::Base.transaction do
+        if voided_by_user_id
+          declaration.voided_by_user_at = Time.current
+          declaration.voided_by_user = User.find(voided_by_user_id)
```

Comment:
`User.find!` or we'll end up with a date and no user, which will be a validation error anyway I guess

Reply:
Doh, of course it does. Fine as-is 👍